### PR TITLE
Shallow and verbose options for git_external

### PR DESCRIPTION
**Pull request by eile**
Squashed and cleaned version of #345

### CODE REVIEW

**Comment by eile** [comment]
@biddisco FYI

File: GitExternal.cmake [hunk 1]
```diff
@@ -8,20 +8,32 @@
...
 #
+# Options which can be supplied to the function:
+#  VERBOSE, when present, this option tells the function to output
+#    information about what operations are being performed by git on
+#    the repo.
+#  SHALLOW, when present, causes a shallow clone of depth 1 to be made
+#    of the specified repo. This may save considerable memory/bandwidth
+#    when only a specific branch of a repo is required and the full history
+#    is not required. Note that the SHALLOW option will only work for a branch
+#    or tag and cannot be used or an arbitrary SHA.
```
Comment (rdumusc) [inline GitExternal.cmake]: typo "be used for"

File: GitExternal.cmake [hunk 2]
```diff
@@ -73,13 +92,29 @@ function(GIT_EXTERNAL DIR REPO TAG)
 
   if(NOT EXISTS "${DIR}")
     # clone
-    message(STATUS "git clone ${REPO} ${DIR}")
+    set(_clone_options --recursive)
+    if(GIT_EXTERNAL_LOCAL_SHALLOW)
+      list(APPEND _clone_options --depth 1 --branch ${TAG})
+    endif()
+    JOIN("${_clone_options}" " " _msg_text)
```
Comment (rdumusc) [inline GitExternal.cmake]: _msg_text is unused

**Comment by rdumusc** [comment]
Changelog is missing, and I see no trace of the "Add macro to rename libs for debug/release builds (on win32)" PR which it is supposed to have integrated... ;-)

**Comment by eile** [comment]
Updated

**Comment by eile** [comment]
This PR was about #345 the lib postfix.

File: CHANGES.md [hunk 3]
```diff
@@ -53,7 +54,7 @@
       a .dmg (if macdeployqt is available)
     * Use C++11 std and stdlib on OSX >= 10.9, C++03 on <= 10.8
 
-### 2014.6
+0# 2014.6
```
Comment (rdumusc) [inline CHANGES.md]: a 0 got lost here

**Comment by rdumusc** [comment]
+1, merge after the 0 is removed!

**Pull request merged**